xarray/testing/assertions.py

```diff
@@ -81,12 +134,19 @@ def assert_equal(a, b):
         assert a.equals(b), formatting.diff_dataset_repr(a, b, "equals")
     elif isinstance(a, Coordinates):
         assert a.equals(b), formatting.diff_coords_repr(a, b, "equals")
+    elif isinstance(a, DataTree):
+        from_root = kwargs.get("from_root", True)
```

Comment:
Would people be more comfortable if I was doing some type checking here for `from_root` being a `bool`?